protocol/msgpack/options.go

```diff
@@ -128,3 +162,66 @@ func (o aggregatedIteratorOptions) SetIteratorPool(value AggregatedIteratorPool)
 func (o aggregatedIteratorOptions) IteratorPool() AggregatedIteratorPool {
 	return o.iteratorPool
 }
+
+func (o aggregatedIteratorOptions) PolicyDecompressionOptions() PolicyDecompressionOptions {
+	return o.policyDecompressionOpts
+}
+
+func (o aggregatedIteratorOptions) SetPolicyDecompressionOptions(policyOpts PolicyDecompressionOptions) AggregatedIteratorOptions {
+	opts := o
+	opts.policyDecompressionOpts = policyOpts
+	return opts
+}
+
+type policyCompressionOptions struct {
```

Comment:
Yep, I actually moved this into the encoder and iterator options so we don't a separate struct for compression.